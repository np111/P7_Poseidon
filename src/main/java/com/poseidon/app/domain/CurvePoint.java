package com.poseidon.app.domain;

import javax.persistence.Entity;
import javax.persistence.Table;


@Entity
@Table(name = "curvepoint")
public class CurvePoint {
    // TODO: Map columns in data table CURVEPOINT with corresponding java fields
}
