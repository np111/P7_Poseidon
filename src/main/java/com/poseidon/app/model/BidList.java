package com.poseidon.app.model;

import java.time.ZonedDateTime;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class BidList {
    private Long id;
    private String account;
    private String type;
    private Double bidQuantity;
    private Double askQuantity;
    private Double bid;
    private Double ask;
    private String benchmark;
    private ZonedDateTime bidListDate;
    private String commentary;
    private String security;
    private String status;
    private String trader;
    private String book;
    private String creationName;
    private ZonedDateTime creationDate;
    private String revisionName;
    private ZonedDateTime revisionDate;
    private String dealName;
    private String dealType;
    private String sourceListId;
    private String side;
}
