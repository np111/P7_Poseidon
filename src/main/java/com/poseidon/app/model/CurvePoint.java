package com.poseidon.app.model;

import java.time.ZonedDateTime;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class CurvePoint {
    private Long id;
    private Long curveId;
    private ZonedDateTime asOfDate;
    private Double term;
    private Double value;
    private ZonedDateTime creationDate;
}
