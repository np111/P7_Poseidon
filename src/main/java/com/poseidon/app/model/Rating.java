package com.poseidon.app.model;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class Rating {
    private Long id;

    @NotBlank(message = "MoodysRating is mandatory")
    private String moodysRating;

    @NotBlank(message = "SandPRating is mandatory")
    private String sandPRating;

    @NotBlank(message = "FitchRating is mandatory")
    private String fitchRating;

    @NotNull(message = "Order is mandatory")
    private Byte orderNumber;
}
