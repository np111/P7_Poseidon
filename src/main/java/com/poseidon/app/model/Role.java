package com.poseidon.app.model;

public enum Role {
    USER,
    ADMIN,
}
