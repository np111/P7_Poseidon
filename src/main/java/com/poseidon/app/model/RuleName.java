package com.poseidon.app.model;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class RuleName {
    private Long id;
    private String name;
    private String description;
    private String json;
    private String template;
    private String sqlStr;
    private String sqlPart;
}
