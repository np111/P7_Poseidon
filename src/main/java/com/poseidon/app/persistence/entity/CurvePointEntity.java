package com.poseidon.app.persistence.entity;

import java.time.ZonedDateTime;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "curvepoint")
@NoArgsConstructor
@Data
public class CurvePointEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "curveId")
    private Long curveId;

    @Column(name = "asOfDate")
    private ZonedDateTime asOfDate;

    @Column(name = "term")
    private Double term;

    @Column(name = "value")
    private Double value;

    @Column(name = "creationDate")
    private ZonedDateTime creationDate;
}
