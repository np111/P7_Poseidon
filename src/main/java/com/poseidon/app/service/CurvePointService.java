package com.poseidon.app.service;

import com.poseidon.app.model.CurvePoint;
import com.poseidon.app.persistence.entity.CurvePointEntity;
import com.poseidon.app.persistence.mapper.CurvePointMapper;
import com.poseidon.app.persistence.repository.CurvePointRepository;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Service;

@RequiredArgsConstructor(onConstructor = @__(@Autowired))
@Service
@Scope("singleton")
public class CurvePointService extends CrudService<CurvePoint, CurvePointEntity, Long> {
    private final @Getter CurvePointRepository repository;
    private final @Getter CurvePointMapper mapper;

    @Override
    protected void setModelId(CurvePoint user, Long id) {
        user.setId(id);
    }
}
