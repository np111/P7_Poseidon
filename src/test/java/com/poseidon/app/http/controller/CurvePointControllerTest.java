package com.poseidon.app.http.controller;

import com.poseidon.app.model.CurvePoint;
import com.poseidon.app.persistence.entity.CurvePointEntity;
import com.poseidon.app.service.CurvePointService;
import lombok.Getter;
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
import org.springframework.boot.test.mock.mockito.MockBean;

@WebMvcTest(CurvePointController.class)
class CurvePointControllerTest extends AbstractCrudControllerTest<CurvePointService, CurvePoint, CurvePointEntity, Long> {
    @MockBean
    private @Getter CurvePointService service;

    public CurvePointControllerTest() {
        super("/curvePoint", CurvePoint.class, Long.class);
    }

    @Override
    protected CurvePoint getModelA() {
        return CurvePoint.builder()
                .id(1L)
                .curveId(2L)
                .term(3D)
                .value(4D)
                .build();
    }

    @Override
    protected CurvePoint getModelB() {
        return CurvePoint.builder()
                .id(3517L)
                .curveId(484607L)
                .term(4.8974702D)
                .value(1.712767D)
                .build();
    }
}
