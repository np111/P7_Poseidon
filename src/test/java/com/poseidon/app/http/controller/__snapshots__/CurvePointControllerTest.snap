com.poseidon.app.http.controller.CurvePointControllerTest.add=[
-
> GET /curvePoint/add
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
""
"  <div class=\"row\">"
"    <h2>Add New Curve Point</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/curvePoint/validate\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"curveId\" class=\"col-sm-2 control-label\">CurveId</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"curveId\" placeholder=\"CurveId\" class=\"col-4\" name=\"curveId\" value=\"\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"term\" class=\"col-sm-2 control-label\">Term</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"term\" placeholder=\"Term\" class=\"col-4\" name=\"term\" value=\"\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"value\" class=\"col-sm-2 control-label\">Value</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"value\" placeholder=\"Value\" class=\"col-4\" name=\"value\" value=\"\">"
"          "
"        </div>"
"      </div>"
""
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/curvePoint/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Add curve point\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
]


com.poseidon.app.http.controller.CurvePointControllerTest.delete=[
-
> GET /curvePoint/delete/1
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <title>Spring Boot</title>"
"</head>"
"<body>"
"<h3>Not Found Exception</h3>"
"<div>"
"  Logged in user: <b class=\"user\"> test-username </b>"
"  <form action=\"/app-logout\" method=\"POST\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"    <input type=\"submit\" value=\"Logout\"/>"
"  </form>"
"</div>"
"<p class=\"error\">Entity not found.</p>"
"</body>"
"</html>"
-
> GET /curvePoint/delete/3517
< HTTP 302
< Content-Language: en
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
< Location: /curvePoint/list
""
]


com.poseidon.app.http.controller.CurvePointControllerTest.list=[
-
> GET /curvePoint/list
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
"  <div class=\"row\">"
"    <div class=\"col-6\">"
"      <a href=\"/bidList/list\">Bid List</a>&nbsp;|&nbsp;"
"      <a href=\"/curvePoint/list\">Curve Points</a>&nbsp;|&nbsp;"
"      <a href=\"/rating/list\">Ratings</a>&nbsp;|&nbsp;"
"      <a href=\"/trade/list\">Trade</a>&nbsp;|&nbsp;"
"      <a href=\"/ruleName/list\">Rule</a>"
"    </div>"
"    <div class=\"col-6 text-right\">"
"      Logged in user: <b class=\"user\"> test-username </b>"
"      <form action=\"/app-logout\" method=\"POST\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"        <input type=\"submit\" value=\"Logout\"/>"
"      </form>"
"    </div>"
"  </div>"
"  <div class=\"row\"><h2>Curve Point List</h2></div>"
"  <div class=\"row\">"
"    <a href=\"/curvePoint/add\" class=\"btn btn-primary btn-sm\">Add New</a>"
"    <table class=\"table table-bordered\">"
"      <thead>"
"      <tr>"
"        <th>Id</th>"
"        <th>CurveId</th>"
"        <th>Term</th>"
"        <th>Value</th>"
"        <th>Action</th>"
"      </tr>"
"      </thead>"
"      <tbody>"
"      <tr>"
"        <td>1</td>"
"        <td>2</td>"
"        <td>3.0</td>"
"        <td>4.0</td>"
"        <td style=\"width: 15%\" class=\"text-center\">"
"          <a href=\"/curvePoint/update/1\">Edit</a>&nbsp;|&nbsp;"
"          <a href=\"/curvePoint/delete/1\">Delete</a>"
"        </td>"
"      </tr>"
"      <tr>"
"        <td>3517</td>"
"        <td>484607</td>"
"        <td>4.8974702</td>"
"        <td>1.712767</td>"
"        <td style=\"width: 15%\" class=\"text-center\">"
"          <a href=\"/curvePoint/update/3517\">Edit</a>&nbsp;|&nbsp;"
"          <a href=\"/curvePoint/delete/3517\">Delete</a>"
"        </td>"
"      </tr>"
"      </tbody>"
"    </table>"
"  </div>"
"</div>"
"</body>"
"</html>"
]


com.poseidon.app.http.controller.CurvePointControllerTest.showUpdate=[
-
> GET /curvePoint/update/1
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <title>Spring Boot</title>"
"</head>"
"<body>"
"<h3>Not Found Exception</h3>"
"<div>"
"  Logged in user: <b class=\"user\"> test-username </b>"
"  <form action=\"/app-logout\" method=\"POST\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"    <input type=\"submit\" value=\"Logout\"/>"
"  </form>"
"</div>"
"<p class=\"error\">Entity not found.</p>"
"</body>"
"</html>"
-
> GET /curvePoint/update/3517
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
"  <div class=\"row\">"
"    <h2>Update CurvePoint</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/curvePoint/update/3517\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"curveId\" class=\"col-sm-2 control-label\">CurveId</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"curveId\" placeholder=\"CurveId\" class=\"col-4\" name=\"curveId\" value=\"484607\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"term\" class=\"col-sm-2 control-label\">Term</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"term\" placeholder=\"Term\" class=\"col-4\" name=\"term\" value=\"4.8974702\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"value\" class=\"col-sm-2 control-label\">Value</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"value\" placeholder=\"Value\" class=\"col-4\" name=\"value\" value=\"1.712767\">"
"          "
"        </div>"
"      </div>"
""
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/curvePoint/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Update curve point\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
]


com.poseidon.app.http.controller.CurvePointControllerTest.update=[
-
> POST /curvePoint/update/1
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
"  <div class=\"row\">"
"    <h2>Update CurvePoint</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/curvePoint/update/1\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"curveId\" class=\"col-sm-2 control-label\">CurveId</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"curveId\" placeholder=\"CurveId\" class=\"col-4\" name=\"curveId\" value=\"\">"
"          <p class=\"text-danger\">CurveId is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"term\" class=\"col-sm-2 control-label\">Term</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"term\" placeholder=\"Term\" class=\"col-4\" name=\"term\" value=\"\">"
"          <p class=\"text-danger\">Term is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"value\" class=\"col-sm-2 control-label\">Value</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"value\" placeholder=\"Value\" class=\"col-4\" name=\"value\" value=\"\">"
"          <p class=\"text-danger\">Value is mandatory</p>"
"        </div>"
"      </div>"
""
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/curvePoint/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Update curve point\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
-
> POST /curvePoint/update/1
< HTTP 302
< Content-Language: en
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
< Location: /curvePoint/list
""
]


com.poseidon.app.http.controller.CurvePointControllerTest.validate=[
-
> POST /curvePoint/validate
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
""
"  <div class=\"row\">"
"    <h2>Add New Curve Point</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/curvePoint/validate\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"curveId\" class=\"col-sm-2 control-label\">CurveId</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"curveId\" placeholder=\"CurveId\" class=\"col-4\" name=\"curveId\" value=\"\">"
"          <p class=\"text-danger\">CurveId is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"term\" class=\"col-sm-2 control-label\">Term</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"term\" placeholder=\"Term\" class=\"col-4\" name=\"term\" value=\"\">"
"          <p class=\"text-danger\">Term is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"value\" class=\"col-sm-2 control-label\">Value</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"value\" placeholder=\"Value\" class=\"col-4\" name=\"value\" value=\"\">"
"          <p class=\"text-danger\">Value is mandatory</p>"
"        </div>"
"      </div>"
""
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/curvePoint/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Add curve point\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
-
> POST /curvePoint/validate
< HTTP 302
< Content-Language: en
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
< Location: /curvePoint/list
""
]